nueljoe/healthcare
# feature(courses):course management an email subscriptions

#### What does this PR do?
Implement course management and email subscriptions

#### Description of Task to be completed?
- implement the `POST /courses` endpoint to create a course.

- implement the `GET /courses` endpoint to fetch a paginated list of courses.

- implement the `GET /courses/:slug` endpoint to fetch a course by it slug.

- implement the `PATCH /courses/:slug` endpoint to update a course.

- implement the `DELETE /courses/:slug` endpoint to delete a course.

- implement the `PATCH /courses/:slug/status` endpoint to publish or unpublish a course.

- refactor the errors utility for better intellisense.

- implement the `GET /courses/categories` endpoint to fetch categories

- implement the `GET /courses/categories/:id/subcategories` to fetch all the subcategories a given category

- implement the `POST /courses/categories` endpoint to create a new category

- implement the `PATCH /courses/categories/:id` endpoint to update a category

- implement the `DELETE /courses/categories/:id` endpoint to delete a course category

- implement the `GET /subscriptions` to fetch a list of subscriptions

- implement the `POST /subscriptions` endpoint to create a new subscription

- implement the `PATCH /subscriptions/:id` endpoint to edit a subscriber's subscription preferences

- implement the `DELETE /subscriptions` endpoint to unsubscribe

#### How should this be manually tested?
- clone the repository
- cd into the project directory and run `cd api` to CD into the `api` directory
- see the README.md file in the `api` directory for the preliminary steps
- ensure that a mysql database in installed and running on your device
- see the postman documentation on `Courses`, `Course Categories` and `Subscriptions`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A
#### Questions:
